contrib/src/instructions/arith.rs

```diff
@@ -0,0 +1,41 @@
+use symbolic_stack_machines_core::{
+    instructions::{AbstractExecRecord, AbstractInstruction, EnvExtension, InstructionResult},
+    memory::Mem,
+    stack::{Stack, StackOpRecord, StackRecord},
+};
+
+pub struct ADD;
+
+impl<T, S, M, Extension, ReturnRecord> AbstractInstruction<S, M, Extension, ReturnRecord> for ADD
+where
+    T: std::ops::Add + std::ops::Add<Output = T> + Clone,
+    S: Stack<StackVal = T>,
+    M: Mem,
+    Extension: EnvExtension,
+{
+    fn exec<C>(
+        &self,
+        stack: &S,
+        _mem: &M,
+        _ext: &Extension,
+    ) -> InstructionResult<AbstractExecRecord<S, M, <Extension as EnvExtension>::DiffRecordType, C>>
+    where
+        C: Into<symbolic_stack_machines_core::constraint::Constraint<C>>,
+    {
+        let mut change_log = AbstractExecRecord::default();
+
+        let op_1: T = stack.peek(0).unwrap();
```

Comment:
Wondering actually how we should handle the case wherein `stack.peek(x)` is `None`. Perhaps this should result in adding a constraint that the values do exist on the stack... Perhaps this already happens when the diff is processed by the machine (so, this is a note-to-self to look into that).

Reply:
I assume there's a standard stack underflow error. I would just throw that for now and assume all initial implementations will have concrete stack size